dandiapi/api/services/embargo/__init__.py

```diff
@@ -76,10 +120,8 @@ def unembargo_dandiset(*, user: User, dandiset: Dandiset):
     if dandiset.uploads.count():
         raise DandisetActiveUploadsError
 
-    # A scheduled task will pick up any new dandisets with this status and email the admins to
-    # initiate the unembargo process
-    dandiset.embargo_status = Dandiset.EmbargoStatus.UNEMBARGOING
-    dandiset.save()
-
-    # Send initial email to ensure it's seen in a timely manner
-    send_dandisets_to_unembargo_message(dandisets=[dandiset])
+    with transaction.atomic():
+        Dandiset.objects.filter(pk=dandiset.pk).update(
+            embargo_status=Dandiset.EmbargoStatus.UNEMBARGOING
+        )
+        transaction.on_commit(lambda: unembargo_dandiset_task.delay(dandiset.pk))
```

Comment:
I haven't looked closely at the PR, but I'm wondering if the lack of [django_capture_on_commit_callbacks](https://pytest-django.readthedocs.io/en/latest/helpers.html#django-capture-on-commit-callbacks) is the cause of problems?

Reply:
That wasn't actually the problem, this part was working correctly (all I needed was `transaction=True` in the test django_db mark). The problem was due to my overall implementation which you pointed out was unnecessary in your other comment.